applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -279,184 +290,176 @@ static int test_tone_button_press(void)
 }
 #endif /* (CONFIG_AUDIO_TEST_TONE) */
 
-/* Handle button activity events */
-static void button_evt_handler(struct button_evt event)
+/* Handle button activity */
+static void button_msg_thread(void)
 {
 	int ret;
+	const struct zbus_channel *chan;
 
-	LOG_DBG("Got btn evt from queue - id = %d, action = %d", event.button_pin,
-		event.button_action);
+	while (!zbus_sub_wait(&button_sub, &chan, K_FOREVER)) {
```

Comment:
The zbus_sub_wait will return 0 if a notification is received, making the while loop continue. It will return:
```
 * @retval -ENOMSG Returned without waiting.
 * @retval -EAGAIN Waiting period timed out.
 * @retval -EINVAL The observer is not a subscriber.
 * @retval -EFAULT A parameter is incorrect, or the function context is invalid (inside an ISR). The
 * function only returns this value when the CONFIG_ZBUS_ASSERT_MOCK is enabled.
```
So I guess it would be better to have:
```suggestion
	while (1) {
	    ret = zbus_sub_wait(&button_sub, &chan, K_FOREVER);
	    ERR_CHK(ret);
```

Reply:
Done